### ISSUE TITLE
Run failure sensor not working

### ISSUE THREAD

**Issue by jkellogg1998** [dagster-io/dagster#19473]
### Dagster version

1.6.1

### What's the issue?

I have a run failure sensor set up to run every 10 minutes and alert in slack if a job run has failed in that time. Up until about a month ago it was working as expected and then it just suddenly stopped working. It started alerting for the SAME group of failed jobs from over a year ago. Every time the sensor ran it would alert for the same failed jobs from a year ago even if a more current job had also failed. It never updates the record_timestamp either to reflect the most recent run of the sensor.

![Screen Shot 2024-01-29 at 11 07 52 AM](https://github.com/dagster-io/dagster/assets/79176641/813a4183-03df-46a7-bd86-7e6d6873862f)

### What did you expect to happen?

I expect the timestamp of the sensor to update every time it's run and for the sensor to only alert for jobs that have failed since the last run time of the sensor. 

### How to reproduce?

```
import os
import datetime
from slack_sdk import WebClient
from slack.errors import SlackApiError
from dagster import RunFailureSensorContext, repository, run_failure_sensor, SkipReason, DagsterInstance

@run_failure_sensor(description='Sends alerts to slack when a dagster job run fails', minimum_interval_seconds=300)
def slack_on_run_failure(context: RunFailureSensorContext):
        try:
           slack_client = WebClient(token=os.environ["SLACK_API_TOKEN"])
           slack_client.chat_postMessage(
                channel="#sc-dagster-alerts",
                text=(
                    f"*Run Failure:* {context.failure_event.message}" +
                    ' \n Check error logs here: ' +
                    'https://scalable-commerce.dagster.cloud/prod/runs/' + 
                    str(context.dagster_run.run_id)
                ),
                icon_emoji=':dagster:'
            )
        except SlackApiError as e:
         #   Handle exceptions for Slack API errors
            print(f"Failed to send Slack message due to {e.response['error']}")

@repository
def my_repository():
    return [
        slack_on_run_failure,
        # Other pipelines and sensors in the repository
    ]
```

this is the code that was working until a month ago. 

### Deployment type

Dagster Cloud

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by alangenfeld** [comment]
This is a `1.6.0` bug that is fixed in the upcoming `1.6.2` release.

**Comment by prha** [comment]
Yes, this is a bug introduced in `1.6.0` that is resolved with `1.6.2`, and only affects Dagster Cloud deployments of run status sensors on code locations running `1.6.0` and `1.6.1`.  Unfortunately, the bug does erroneously persist the cursor value, so you might need to clear out those values (using the cursor `Edit` button in the sensor page UI) if you have operated the run status sensor using code from one of the problem versions.  

If you have upgraded directly from `<1.6.0` to `>=1.6.2`, you will not have to do anything.
